api/mesh/v1alpha1/gateway_route.proto

```diff
@@ -226,6 +226,13 @@ message MeshGatewayRoute {
         Redirect redirect = 3;
         Rewrite rewrite = 4;
         HeaderFilter response_header = 5;
+        // Option to indicate that during forwarding, the host header should be
+        // swapped with the hostname of the upstream host chosen by the Envoy's
+        // cluster manager.
+        // BE AWARE:
+        // - it's mutually exclusive with request_header filter which explicitly
+        //   replaces "host" header
+        bool rewrite_host_to_backend_hostname = 6;
```

Comment:
I had a bad feeling about a top level `bool` here but wasn't sure where to put it. I think it can go under `rewrite` so that we have:

```
- rewrite:
    hostHeaderToBackendHostname: true
```

Reply:
done